src/Adapter/ReactPhp/EventLoop.php

```diff
@@ -181,7 +190,14 @@ public function promiseRejected(\Throwable $throwable): Promise
      */
     public function idle(): Promise
     {
-        $deferred = $this->deferred();
+        /** @var Deferred $deferred */
+        $deferred = null;
+
+        $deferred = $this->deferred(
+            function () use (&$deferred) {
+                $deferred->reject(new \M6Web\Tornado\CancelledException());
+            }
+        );
```

Comment:
Looks strange… this code means that when the promise of a deferred is cancelled you want to reject the deferred to… reject the promise that you cancelled 🤨
According to the documentation, it's not possible to cancel `futureTick` in ReactPhp… to ensure consistent behavior, we could use a timer with a 0 delay 🤔